brightray/browser/linux/libnotify_notification.cc

```diff
@@ -126,6 +128,20 @@ void LibnotifyNotification::Show(const NotificationOptions& options) {
         notification_, "x-canonical-append", "true");
   }
 
+  // Send the desktop name to identify the application
+  // The desktop-entry is the part before the .desktop
+  std::unique_ptr<base::Environment> env(base::Environment::Create());
+  std::string desktop_id = libgtkui::GetDesktopName(env.get());
+  if (!desktop_id.empty()) {
+    std::size_t last_pos = desktop_id.find_last_of(".desktop");
+    if (last_pos != std::string::npos) {
+      desktop_id = desktop_id.substr(0, last_pos);
+    }
+
+    libnotify_loader_.notify_notification_set_hint_string(
+        notification_, "desktop-entry", desktop_id.c_str());
+  }
```

Comment:
I hadn't looked into this before, so some breadcrumbs on why this ought to work....

`libgtkui::GetDesktopName()` is implemented [this way](https://cs.chromium.org/chromium/src/chrome/browser/ui/libgtkui/gtk_util.cc?q=GetDesktopName&sq=package:chromium&dr=CSs&l=122) in `chrome/browser/ui/libgtkui/gtk_util.cc`:

```c++
std::string GetDesktopName(base::Environment* env) {
#if defined(GOOGLE_CHROME_BUILD)
  return "google-chrome.desktop";
#else  // CHROMIUM_BUILD
  // Allow $CHROME_DESKTOP to override the built-in value, so that development
  // versions can set themselves as the default without interfering with
  // non-official, packaged versions using the built-in value.
  std::string name;
  if (env->GetVar("CHROME_DESKTOP", &name) && !name.empty())
    return name;
  return "chromium-browser.desktop";
#endif
}
```

and `CHROME_DESKTOP` is [set](https://github.com/electron/electron/blob/master/atom/browser/api/atom_api_app.cc) by Electron apps:

```c++

void App::SetDesktopName(const std::string& desktop_name) {
#if defined(OS_LINUX)
  std::unique_ptr<base::Environment> env(base::Environment::Create());
  env->SetVar("CHROME_DESKTOP", desktop_name);
#endif
}
```

And, though it's probably common knowledge to people who might be reading this :smile: , `desktop-entry` is documented in [this list of fdo hints at developer.gnome.org](https://developer.gnome.org/notification-spec/).

```text
This specifies the name of the desktop filename representing the calling program. This should be the same as the prefix used for the application's .desktop file. An example would be "rhythmbox" from "rhythmbox.desktop". This can be used by the daemon to retrieve the correct icon for the application, for logging purposes, etc.
```

tl;dr: lgtm